components/provider_cldr/src/transform/dates.rs

```diff
@@ -115,6 +115,83 @@ impl From<&cldr_json::StylePatterns> for gregory::patterns::StylePatternsV1 {
     }
 }
 
+impl From<&cldr_json::DateTimeFormats> for gregory::patterns::DateTimeFormatsV1 {
+    fn from(other: &cldr_json::DateTimeFormats) -> Self {
+        use gregory::patterns::{
+            PatternV1, SkeletonFieldsV1, SkeletonFieldsV1Error, SkeletonTupleV1,
+        };
+
+        // TODO(#308): Support numbering system variations. We currently throw them away.
+        Self {
+            style_patterns: gregory::patterns::StylePatternsV1 {
+                full: other.full.get_pattern().clone(),
+                long: other.long.get_pattern().clone(),
+                medium: other.medium.get_pattern().clone(),
+                short: other.short.get_pattern().clone(),
+            },
+            skeletons: {
+                let mut skeleton_tuples: Vec<SkeletonTupleV1> = Vec::new();
+
+                // The CLDR keys for available_formats can have duplicate skeletons with either
+                // an additional variant, or with multiple variants for different plurals.
+                for (skeleton_str, pattern_str) in other.available_formats.0.iter() {
+                    let mut unique_skeleton = String::new();
+                    let mut variant_text = String::new();
+
+                    // Split out the unique skeleton component, and the variant text.
+                    //   "MMMMW-count-two"
+                    //   unique_skeleton: "MMMMW"
+                    //   variant_text: "-count-two"
+                    let chars = skeleton_str.chars();
```

Comment:
It's nicer to use `skeleton_str.split('-')` and the pull items off that iterator.  You could also consider using itertools' `next_tuple()` and expand it into a tuple like I do in #541.

Reply:
I opted for the split implementation, but not the `next_tuple` as it would add dependencies and include an assumption that variant text would always have a fixed number of splits.

Reply:
This is: dd688d8efee45053a50789db64d90eb753fe22a2